Assets/Huawei/Scripts/ML/LangDetector/HMSMLLanguageDetectionManager.cs

```diff
@@ -0,0 +1,180 @@
+using HmsPlugin;
+using HuaweiMobileServices.ML.Common;
+using HuaweiMobileServices.ML.LanguageDetection;
+using HuaweiMobileServices.ML.LanguageDetection.Cloud;
+using HuaweiMobileServices.ML.LanguageDetection.Local;
+using HuaweiMobileServices.Utils;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+namespace Assets.Huawei.Scripts.ML.LangDetector
+{
+    public enum DetectionMode
+    {
+        Online = 0,
+        Offline = 1
+    }
+    public class HMSMLLanguageDetectionManager : HMSManagerSingleton<HMSMLLanguageDetectionManager>
+    {
+
+        private const string TAG = "[HMS] HMSMLLanguageDetectionManager ";
+        private readonly string apiKey;
+        private readonly int DefaultUserRegion = MLApplication.REGION_DR_GERMAN;
+        private DetectionMode currentDetectionMode;
+
+        private MLRemoteLangDetectorSetting remoteSetting;
+        private MLRemoteLangDetector remoteDetector;
+        private MLLocalLangDetectorSetting localSetting;
+        private MLLocalLangDetector localDetector;
+
+        private bool IsLangDetectionModuleEnabled => HMSMLKitSettings.Instance.Settings.GetBool(HMSMLKitSettings.EnableLanguageDetectionModule);
+
+        public HMSMLLanguageDetectionManager()
+        {
+            if (!IsLangDetectionModuleEnabled)
+            {
+                Debug.Log($"{TAG} -> Lang Detection Module is not enabled");
+                return;
+            }
+            apiKey = HMSMLKitSettings.Instance.Settings.Get(HMSMLKitSettings.MLKeyAPI);
+            HMSManagerStart.Start(OnAwake, TAG);
+        }
+        private void OnAwake()
+        {
+            Initialize();
+        }
+        private void Initialize()
+        {
+            var instance = MLApplication.Initialize();
+            instance.SetApiKey(apiKey);
+            instance.SetUserRegion(DefaultUserRegion);
+            Debug.Log($"{TAG} -> Initialized");
+        }
+
+        public void Configure(DetectionMode detectionMode)
+        {
+            currentDetectionMode = detectionMode;
+            switch (detectionMode)
+            {
+                case DetectionMode.Online:
+                    SetOnlineDetection();
+                    break;
+                case DetectionMode.Offline:
+                    SetOfflineDetection();
+                    break;
+            }
+        }
+
+
+        private void SetOnlineDetection()
+        {
+            float f = 0.01f;
```

Comment:
If the user wants to change this parameter from their own game scene, it should be able to be changed by sending the parameter from their own game scene. Like;

private void SetOnlineDetection(float trustedThreshold = 0.01f)
        {
            ....
          }